week2/index.js

```diff
@@ -0,0 +1,128 @@
+"use strict";
+
+const fs = require('fs');
+
+const TODOS_FILENAME = process.env["TODOS_FILENAME"] || "to-do-list.json";
+
+let myTodoList = []
+
+
+// Helper functions definitions
+
+let readFileFunction = (filename) => {
+    return new Promise((resolve, reject) => {
+        fs.readFile(filename, "utf8", (err, data) => {
+            if (err) {
+                if (err.code === "ENOENT") {
+                    // No such file or directory
+                    console.log("I'm sorry, friend, but that file doesn't exist.");
+                } else if (err.code === "EACCES") {
+                    console.log("That file exists but you don't have permission.");
+                } else {
+                    console.log(`Uh oh, an unknown error occured: ${err.message}`);
+                }
+                process.exit()
+            }
+            //  console.log(`Successfully read the file: ${filename}`)
+            resolve(data)
+        })
+    })
+};
+
+let writeFileFunction = (filename, dataAsString) => {
+    return new Promise((resolve, reject) => {
+        fs.writeFile(filename, dataAsString, (err) => {
+            if (err) reject(console.log(`error happened while writing the file \n${err.message}`));
+            //  console.log(`${filename} is saved`);
+            resolve();
+        })
+    })
+}
+
+let addingTask = (taskText) => {
+    readFileFunction(TODOS_FILENAME).then((data) => {
+        data = data || "[]"
+        myTodoList = JSON.parse(data)
+        myTodoList.push({ task: taskText });
+        writeFileFunction(TODOS_FILENAME, JSON.stringify(myTodoList)).then(() => console.log("Done adding a task."));
+    })
+}
+
+let showHelpMenu = () => {
+    readFileFunction("helpMenu.txt").then((data) => console.log(data))
+}
+
+let resetFunction = () => {
+    myTodoList = [];
+    writeFileFunction(TODOS_FILENAME, "")
+    console.log("The tasks list has been reset")
```

Comment:
since `writeFileFunction` returns a promise, your `console.log` will be called before the file has finished writing. You probably want to use `.then()` and `.catch()` here.